app/api/score_sub.py

```diff
@@ -216,7 +216,16 @@ async def submit_score(
         else:
             score.status = ScoreStatus.FAILED
 
-        score.time_elapsed = score_time if score.passed else fail_time
+        logging.warning(
+            "Score time output",
+            extra={
+                "user_id": score.user_id,
+                "score_time": score_time,
+                "fail_time": fail_time,
+            },
+        )
+
+        score.time_elapsed = score_time
```

Comment:
Removed fail time, which turns out to be a metric for the time through a map at which a player failed, not the amount of time the player spent on the map.

e.g. on the unforgiving "Stairway to the skies", if you fail 20 seconds into the map, your score_time will be 20k while the fail_time will be 2926k (the offset into the map).

Reply:
New proposal:
```py
class TimePlayed(TypedDict):
    seconds_active: float
    seconds_inactive: float


def adjust_duration_for_mods(duration: float, mods: int) -> float:
    if mods & (Mods.DOUBLETIME | Mods.NIGHTCORE):
        duration /= 1.50  # 50% speed increase
    elif mods & Mods.HALFTIME:
        duration /= 0.75  # 25% speed decrease
    return duration


def calculate_time_played(
    *,
    failed: bool,
    score_time: int,
    fail_time: int,
    hit_length: int,
    first_hitobject_offset: int,
    mods: int,
) -> TimePlayed:
    if failed:
        seconds_active = fail_time - first_hitobject_offset
    else:
        seconds_active = hit_length

    seconds_inactive = score_time - seconds_active

    return {
        "seconds_active": adjust_duration_for_mods(seconds_active, mods),
        "seconds_inactive": adjust_duration_for_mods(seconds_inactive, mods),
    }
```

And the addition of 
```sql
alter table beatmaps add column first_hitobject_offset int not null;
```

Reply:
I'll implement this in a future set of PRs